textproto/header.go

```diff
@@ -440,7 +524,16 @@ func ReadHeader(r *bufio.Reader) (Header, error) {
 			return newHeader(fs), fmt.Errorf("message: malformed MIME header line: %v", string(kv))
 		}
 
-		key := textproto.CanonicalMIMEHeaderKey(string(trim(kv[:i])))
+		key_bytes := trim(kv[:i])
+
+		// Verify that there are no invalid characters in the header key.
+		for _, c := range key_bytes {
+			if !validHeaderFieldByte(c) {
+				return newHeader(fs), fmt.Errorf("message: malformed MIME header line: %v", string(kv))
```

Comment:
Can we just print the header key here (without the value)? I'd prefer not to have `\r` and `\n` in error messages.